csrc/scheduler/utils.cpp

```diff
@@ -170,21 +170,32 @@ std::optional<size_t> mergeDims(
   if (to_merge.size() == 1) {
     return to_merge[0];
   }
-  std::sort(to_merge.begin(), to_merge.end());
```

Comment:
Why this is changed?

Reply:
We can't blindly sort and merge with the sorted order.

i.e. The transform is applied on reference1 and it needs to be populated to group2 for tiling.
if we have reference1 (i0, i1, i2, i3, i4) -> reference2 (i3, i4, i2, i0, i1).

Consider a split on i2 that gives us (i2o, i2i). So reference1 becomes (i0, i1, i2o, i2i, i3, i4) and dims_merged_with_2 is `i0, i2o`.
previously we have merge i0, i2o, and then merge i1 with them. Which gives us. (i0 * i2o * i1). This is not vectorizable on reference2 (i3, i4, i2, i0, i1).
So we need to ensure that we merge things in the logical order as they show up in `dims_merged_with_2`.

Reply:
https://github.com/NVIDIA/Fuser/issues/675#issuecomment-1671627684

Reply:
Thank you for the explanation. Could you also pull this out as a separate PR? I don't like packing fixes on multiple issues in the same PR. Also, in this case, does it make sense to make the `to_merge` argument an `unordered_set`? It would also be good to modify `FusionMergeDims_CUDA` to test this behavior change.

Reply:
Will do (regarding split out to a new PR).

Meanwhile, we do want to keep std::vector instead of std::unordered_set, since we can't dictate an iterating order with unordered_set.